File: src/terminal/terminal_ffi.erl
-module(terminal_ffi).

-export([enter_raw/0, window_size/0]).

enter_raw() ->
    shell:start_interactive({noshell, raw}).

window_size() ->
    case io:columns() of
        {ok, Cols} ->
            case io:rows() of
                {ok, Rows} ->
                    {Cols, Rows};
                _ ->
                    error
            end;
        _ ->
            error
    end.

File: src/terminal/terminal_ffi.mjs
export function enter_raw() {
  process.stdin.setRawMode(true);
}

export function exit_raw() {
  process.stdin.setRawMode(false);
}

export function is_raw_mode() {
  return process.stdin.isRaw;
}

export function window_size() {
  return [process.stdout.columns, process.stdout.rows];
}
